herddb-core/src/main/java/herddb/core/TableManager.java

```diff
@@ -3014,11 +3017,13 @@ private TableCheckpoint checkpoint(
             result = new TableCheckpoint(table.name, sequenceNumber, actions);
 
             end = System.currentTimeMillis();
-
-            LOGGER.log(Level.INFO, "checkpoint {0} finished, logpos {1}, {2} active pages, {3} dirty pages, "
-                            + "flushed {4} records, total time {5} ms",
-                    new Object[]{table.name, sequenceNumber, pageSet.getActivePagesCount(),
-                            pageSet.getDirtyPagesCount(), flushedRecords, Long.toString(end - start)});
+            long totalTime = end - start;
+            if (totalTime > 100) {
```

Comment:
Why logging based only on checkpoint duration?
Avoid to print a fast checkpoint or an empty checkpoint?
Print only if didn't flush records?

Reply:
this is in order to not print useless checkpoints, like in "local" mode.
also it is useful to see if a checkpoint blocked table operations for a significant period of time.

In my opinion it is still good to see when checkpoints happen, even if much time passed since we saw checkpoint related problems.

using the number of records as threshold may result in hiding problems (the checkpoint is not doing anything) and probably it would also generate spam in "local mode" if the table has more than one dirty records

Do you have any suggestions ?

Reply:
I agree that is useful to know when checkpoint happens and how much time it took. My question was exactly about it because such check won't print checkpoint data if faster than 100 ms, it smells a little.
If we want to avoid useless checkpoint print it should't be logged only if it was empty (and perhaps fast too). Still in such case could be helpful a debug line

Reply:
agreed.
I have checked the condition
PTAL @diegosalvi